superset/connectors/sqla/models.py

```diff
@@ -834,6 +843,10 @@ def get_sqla_query(  # sqla
                         where_clause_and.append(col_obj.get_sqla_col() == None)
                     elif op == "IS NOT NULL":
                         where_clause_and.append(col_obj.get_sqla_col() != None)
+
+        where_clause_and += [
+            text("({})".format(f)) for f in self._get_row_level_filters()
```

Comment:
I think sqlalchemy should be doing the right thing here around parenthesis, so no need to add them explicitely
https://docs.sqlalchemy.org/en/13/core/sqlelement.html#sqlalchemy.sql.expression.BinaryExpression.self_group

Something like 
```python
where_clause_and += [text(f) for f in self._get_row_level_filters()]
```
Oh and I guess the `text()` call could be in the method, could rename to `_get_sqla_row_level_filters`, so it's just `where_clause_and += self._get_row_level_filters()`

Reply:
I tested this by adding an RLS filter that contained an `OR`.  It didn't end up getting wrapped in parentheses. So I still have the `"({})".filter(f)` in there, but I've moved it to the `_get_sqla_row_level_filters` method.
It looks like similar was done in the `get_sql_query()` method, presumably for similar reason
https://github.com/apache/incubator-superset/blob/cae0583e0fbf8b60b9ee86179b4defef5da9b2c2/superset/connectors/sqla/models.py#L841